core/src/game_state.rs

```diff
@@ -1480,7 +1504,20 @@ impl DrawPhase {
             bail!("nobody has bid yet")
         } else {
             let winning_bid = bail_unwrap!(self.autobid.or_else(|| self.bids.last().copied()));
-            let landlord = self.propagated.landlord.unwrap_or(winning_bid.id);
+            let first_bid = bail_unwrap!(self.autobid.or_else(|| self.bids.first().copied()));
+            let landlord = match self.propagated.first_landlord_selection_policy {
+                FirstLandlordSelectionPolicy::ByWinningBid => {
+                    self.propagated.landlord.unwrap_or(winning_bid.id)
+                }
+                FirstLandlordSelectionPolicy::ByFirstBid => {
+                    if self.propagated.landlord.is_none() {
```

Comment:
I don’t think you need this if statement— unwrap_or only uses the right hand side when the left hand side is None